src/main.rs

```diff
@@ -337,6 +337,15 @@ fn main() -> anyhow::Result<()> {
         log::info!("Telemetry reporting disabled");
     }
 
+    // Helper to better log start errors
+    let log_err = |server_name, result| match result {
+        Err(err) => {
+            log::error!("Error while starting {} server: {}", server_name, err);
+            Err(err)
+        }
+        ok => ok,
+    };
```

Comment:
```suggestion
    let log_err = |server_name, result| {
        if let Err(err) = &result {
            log::error!("Error while starting {} server: {}", server_name, err);
        }
        result
    };
```

Also we may want to rename `log_err` to `log_err_if_any` or `with_log_err` because it doesn't just log the error, but also pass through the result. Otherwise the callers look a bit weird to me.